pkg/capture/capture_manager.go

```diff
@@ -560,17 +572,17 @@ func (cm *Manager) logErrors(ctx context.Context, iface string, errsChan <-chan
 
 				// Ensure there is no conflict with calls to update() that might already be
 				// taking down this interface
-				cm.Lock()
-				defer cm.Unlock()
+				cm.captures.Lock()
```

Comment:
This is a tiny detail, but important. We lock the set of captures (because that's the only thin we potentially manipulate here) instead of the whole manager.